### PR TITLE
JSUI-2547 Refactor DynamicFacets analytics meta 

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2547

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 1]
```diff
@@ -1282,12 +1281,14 @@ export var analyticsActionCauseList = {
    * `actionType`: `'dynamicFacet'`
    *
    * Logging an event with this actionType also adds the following key-value pairs in the custom data property of the Usage Analytics HTTP service request.
```
Comment (fbeaudoincoveo) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: Suggestion:

```
The required and optional properties of an [`IAnalyticsDynamicFacetMeta`](@link IAnalyticsDynamicFacetMeta) object are added as custom data when logging a usage analytics event matching this `actionCause`/`actionType`.
```

Comment (fbeaudoincoveo) [reply src/ui/Analytics/AnalyticsActionListMeta.ts]: We can use the exact same thing for dynamicFacetDeselect

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 2]
```diff
@@ -1318,8 +1321,10 @@ export var analyticsActionCauseList = {
...
```
Comment (fbeaudoincoveo) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: Suggestion:
```
The required properties of an [`IAnalyticsDynamicFacetMeta`](@link IAnalyticsDynamicFacetMeta) object are added as custom data when logging a usage analytics event matching this `actionCause`/`actionType`.
```

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 3]
```diff
@@ -429,10 +430,21 @@ export class DynamicFacet extends Component implements IAutoLayoutAdjustableInsi
     this.padding && this.padding.pin();
   }
 
+  // Complete facet analytics meta
   public get analyticsFacetState(): IAnalyticsDynamicFacetMeta[] {
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply src/ui/DynamicFacet/DynamicFacet.ts]: We previously decided not to

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 4]
```diff
@@ -135,15 +135,14 @@ export enum AnalyticsDynamicFacetType {
 }
 
 export interface IAnalyticsDynamicFacetMeta {
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply src/ui/Analytics/AnalyticsActionListMeta.ts]: valuePosition / facetPosition are 1-based 
facetPosition accounts for dynamic facets in the page

[the remark this answers is not in the document]
Comment (fbeaudoincoveo) [reply src/ui/Analytics/AnalyticsActionListMeta.ts]: Thanks! I'll let you make the necessary adjustments in the doc :)

**Pull request merged**